scipy/stats/stats.py

```diff
@@ -4226,7 +4236,15 @@ def spearmanr(a, b=None, axis=0, nan_policy='propagate'):
         # errors before taking the square root
         t = rs * np.sqrt((dof/((rs+1.0)*(1.0-rs))).clip(0))
 
-    prob = 2 * distributions.t.sf(np.abs(t), dof)
+    if alternative == 'less':
+        prob = distributions.t.cdf(t, dof)
+    elif alternative == 'greater':
+        prob = distributions.t.sf(t, dof)
+    elif alternative == 'two-sided':
+        prob = 2 * distributions.t.sf(np.abs(t), dof)
+    else:
+        raise ValueError("alternative should be "
+                         "'less', 'greater' or 'two-sided'")
```

Comment:
@WarrenWeckesser this is currently copy-pasted per our understanding of your suggestion. Should we make a function? Should we use the function written for the t-tests? Should we generalize that function to any distribution, where `distributions.t` would just be replaced a `distribution` parameter?

Reply:
@DominicChm updated `_ttest_finish` in gh-12597, so yeah, let's use that where appropriate.